ingest/ingest_files.py

```diff
@@ -10,32 +10,38 @@
 import re
 from itertools import islice
 
+import pandas as pd
+
 
 class IngestFiles:
-    def __init__(self, file_path, allowed_file_types, is_MTX=False):
+    def __init__(self, file_path, allowed_file_types, *, is_MTX=False, open_as=None):
         if not os.path.exists(file_path):
             raise IOError(f"File '{file_path}' not found")
         self.allowed_file_types = allowed_file_types
-        self.file_type, self.file = self.open_file(file_path)
+        self.file_type, self.file = self.open_file(file_path, open_as)
         # Keeps tracks of lines parsed
         self.amount_of_lines = 0
         self.is_MTX = is_MTX
 
-    def open_file(self, file_path):
+    def open_file(self, file_path, open_as=None):
         """ Opens txt, csv, or tsv formatted files"""
         open_file = open(file_path, encoding='utf-8-sig')
         file_connections = {
             # Remove BOM with encoding='utf-8-sig'
-            'text/csv': self.open_csv(open_file),
+            'text/csv': self.open_csv,
             'text/plain': open_file,
-            'text/tab-separated-values': self.open_tsv(open_file),
+            'text/tab-separated-values': self.open_tsv,
+            'pandas': self.open_pandas
```

Comment:
Suggestion until the future refactoring you mentioned:

```suggestion
            'dataframe': self.open_pandas
```